pkg/validate/raw/validate/source_spec_validator.go

```diff
@@ -142,6 +142,10 @@ func OciSpec(oci *v1beta1.Oci, rs client.Object) status.Error {
 	// will fail to apply.
 	switch oci.Auth {
 	case configsync.AuthGCENode, configsync.AuthNone:
+	case configsync.AuthK8sServiceAccount:
+		if oci.GCPServiceAccountEmail != "" {
+			return IllegalGCPSAEmail(v1beta1.OciSource, rs)
```

Comment:
It seems like this should return IllegalGCPSAEmail for _any_ auth type that is _not_ GCPServiceAccountEmail, but this only returns for AuthK8sServiceAccount. Is that intentional?

Reply:
The presence of GCPServiceAccountEmail will interfere with the  AuthK8sServiceAccount auth type, but not other auth types.
The check is a MUST for  AuthK8sServiceAccount, but I agree it is more consistent if other auth types have the same check.

Reply:
It turns out a bigger change, and I extracted into a separate PR: #1122.